src/main/java/org/threadly/litesockets/buffers/MergedByteBuffers.java

```diff
@@ -0,0 +1,316 @@
+package org.threadly.litesockets.buffers;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+
+/**
+ * This class is used to combine multiple ByteBuffers into 1 simplish to use interface.
+ * It provides most of the features of a single ByteBuffer, but with the ability to perform those 
+ * operations spanning many ByteBuffers.
+ * 
+ * The idea here is to keep from having to copy around and merge ByteBuffers as much as possible. 
+ * 
+ * NOTE: This is not threadSafe.  It should only be accessed by 1 thread at a time.
+ * 
+ */
+public interface MergedByteBuffers {
+  int BYTES_IN_LONG = Long.SIZE/Byte.SIZE;
+  int BYTES_IN_INT = Integer.SIZE/Byte.SIZE;
+  int BYTES_IN_SHORT = Short.SIZE/Byte.SIZE;
+
+  short UNSIGNED_BYTE_MASK = 0xff;
+  int UNSIGNED_SHORT_MASK = 0xffff;
+  long UNSIGNED_INT_MASK = 0xffffffffL;
+  
+  ByteBuffer EMPTY_BYTEBUFFER = ByteBuffer.allocate(0);
+
+  
+  /**
+   * This method allows you to add ByteBuffers to the MergedByteBuffers.  
+   * All must be done in order of how you want to pull the data back out.
+   * 
+   * @param ba - The byte[] to add to the MergedByteBuffers
+   */
+  public void add(final byte[] ...ba);
+  
+  /**
+   * This method allows you to add ByteBuffers to the MergedByteBuffers.  
+   * All must be done in order of how you want to pull the data back out.
+   * 
+   * @param buffer - The byte buffer to add to the MergedByteBuffers
+   */
+  public void add(final ByteBuffer ...buffer);
+  
+  /**
+   * This method allows you to add a MergedByteBuffers to another MergedByteBuffers.  
+   * All must be done in order of how you want to pull the data back out.
+   * 
+   * @param mbb - The MergedByteBuffers to put into this MergedByteBuffers
+   */
+  public void add(final MergedByteBuffers ...mbb);
+
+  /**
+   * Make a complete copy of this MergedByteBuffer.  Both references should function independently, but
+   * they are still using the same ByteBuffer backing arrays so any change to the actual byte[] in the 
+   * backing ByteBuffers will change in both.
+   * 
+   * @deprecated changed to {@link #duplicate()}
```

Comment:
trying to make it work more like normal ByteBuffer objects, but I do get your point.  On that do you think there is a reason to have/add a deep copy?